### PR TITLE
Use async in download wallet

### LINKED ISSUE #126: Generate Wallet creates empty File
A regression in the generate wallet flow is causing files to be downloaded as empty objects.

### PR DESCRIPTION
**Pull request by skubakdj**
Fixes #126.

### CODE REVIEW

File: common/containers/Tabs/GenerateWallet/components/DownloadWallet.jsx [hunk 1]
```diff
@@ -27,11 +27,16 @@ export default class DownloadWallet extends Component {
   }
 
   componentWillMount() {
```
Comment (crptm) [inline common/containers/Tabs/GenerateWallet/components/DownloadWallet.jsx]: async componentWillMount

**Comment by crptm** [comment]
Before fixing a bug we need to either introduce tests or fix type annotation to break flow

File: common/containers/Tabs/GenerateWallet/components/DownloadWallet.jsx (hunk 1 above)
Comment (skubakdj) [inline common/containers/Tabs/GenerateWallet/components/DownloadWallet.jsx]: I tried something like
```
  async componentWillMount() {
    if (!this.props.wallet) return;
    const keystore = await this.props.wallet.toKeystore(this.props.password);
    this.setState({ keystore });
  }
```
which seem to work when using the app, but flow threw an error. Seems to be a flow limitation?

https://github.com/facebook/flow/issues/1803

File: common/libs/keystore.js [hunk 2]
```diff
@@ -10,6 +10,13 @@ import { sha3, privateToAddress } from 'ethereumjs-util';
 import scrypt from 'scryptsy';
 import uuid from 'uuid';
 
+export type UtcKeystore = {
+  version: number,
+  id: string,
+  address: string,
+  Crypto: Object
+};
+
```
Comment (skubakdj) [inline common/libs/keystore.js]: Is a single layer deep sufficient for our purposes, or should all of `Crypto` be typed out?

File: common/containers/Tabs/GenerateWallet/components/DownloadWallet.jsx (hunk 1 above)
Comment (crptm) [inline common/containers/Tabs/GenerateWallet/components/DownloadWallet.jsx]: Okay then

File: common/libs/keystore.js (hunk 2 above)
Comment (crptm) [inline common/libs/keystore.js]: We can live with it

File: common/containers/Tabs/GenerateWallet/components/DownloadWallet.jsx [hunk 3]
```diff
@@ -27,16 +34,23 @@ export default class DownloadWallet extends Component {
...
-    this.keystore = this.props.wallet.toKeystore(this.props.password);
+    if (!this.props.wallet) return;
```
Comment (crptm) [inline common/containers/Tabs/GenerateWallet/components/DownloadWallet.jsx]: Prop definition is incorrect if this could be true

**Comment by dternyak** [comment]
@crptm Are you good with this being merged? It looks good on my side.

**Comment by crptm** [comment]
🚢

**Pull request merged**